pkg/cmd/init/exec.go

```diff
@@ -219,7 +219,13 @@ func (o *Options) run() error {
 			}
 		}
 
-		//if service-account wait for the sa secret
+		//  apply an empty image pull secret if the image pull secret is not found in the ns.
+		err = helpers.ApplyImagePullSecret(context.TODO(), "open-cluster-management", kubeClient)
```

Comment:
if it is dry run, will we create the secret?

Reply:
I think we need to create the secret . have fixed.